pkg/cmd/roachtest/main_test.go

```diff
@@ -167,6 +172,188 @@ func Test_updateSpecForSelectiveTests(t *testing.T) {
+		skippedTestsInSpec := 0
+		for _, s := range specs {
+			if s.Skip == getSkippedMessage(s.Name) {
+				skippedTestsInSpec++
+			}
+			// if the test is present in dataMap and snowflake response has "last_failure_is_preempt=true", the same
+			// must be marked in the spec
+			if d, ok := dataMap[s.Name]; ok && d[testselector.DataLastPreempted] == "yes" {
+				require.True(t, s.IsLastFailurePreempt())
+			} else {
+				require.False(t, s.IsLastFailurePreempt())
+			}
+			// if the test is present in dataMap and snowflake response and the test is skipped by test selector
+			// the entry dataMap must be "selected=no"
+			if d, ok := dataMap[s.Name]; ok && s.Skip == "test selector" {
+				require.Equal(t, "no", d[testselector.DataSelectedIndex])
+			}
+			// if a test is in toBeSelectedTestsMap, this test has to be selected by test selector unless it is skipped already
+			// So, the Skip is either blank or has a value as "<test name> skipped"
+			if _, ok := toBeSelectedTestsMap[s.Name]; ok {
+				allTestsToBeSelected++
+				assert.True(t, s.Skip == "" || s.Skip == getSkippedMessage(s.Name), s.Skip)
+			}
+		}
+		require.Equal(t, countOfSpecSkippedTests, skippedTestsInSpec)
+		require.Equal(t, allTestsToBeSelected, len(toBeSelectedTestsMap))
+	})
+}
+
+func getSkippedMessage(testName string) string {
+	return fmt.Sprintf("%s skipped", testName)
+}
+
+// getRandomisedTests returns "totalTests" number of randomised tests.
+// Out of the totalTests,
+// > 0-90% of the tests are added as tests in specs
+// > 10-100% of the tests are added as snowflake returned tests
+// So, the overlap is 80%
+// The above tests are shuffled and randomly added different conditions.
+// The function returns:
+// > List of test specs
+// > The rows to be returned by the snowflake as [][]string
+// > The rows as *sqlmock.Rows
+// > The map of tests that will be selected based on the percent criteria
+func getRandomisedTests(
+	t *testing.T, totalTests int,
+) ([]registry.TestSpec, [][]string, *sqlmock.Rows, map[string]struct{}) {
+	testNames, err := generateUniqueStrings(totalTests, 10, 100)
+	require.Nil(t, err)
+	// As mentioned,
+	// 0% to 90% of tests are added to test spec
+	// 10% to 100% of tests are added to snowflake tests
+	// So, the overlap is 80%
+	sfStart := int(float32(len(testNames)) * 0.1)
+	specEnd := int(float32(len(testNames)) * 0.9)
+	// testForSF are the name of tests which are returned by SF query
+	testForSF := shuffleStrings(t, testNames[sfStart:])
+	// testForSpecs are the name of tests which are in the test spec
+	testForSpecs := shuffleStrings(t, testNames[:specEnd])
+	// commonSuccessTestMap contains all test that are present in both specs and snowflake minus the tests that are
+	// marked as "selected=true"
+	commonSuccessTestMap := make(map[string]struct{})
+	for _, tn := range testNames[sfStart:specEnd] {
+		// first add all the common tests. The tests marked as "selected=true" will be removed later
```

Comment:
small nit: Our style guide recommends making non inline comments full sentences. Just slightly easier to read IMO.